### ISSUE TITLE
Travis --> GHA

### ISSUE THREAD

**Issue by jGaboardi** [pysal/spreg#56]
Prerequisite for solving #51 

**To Do**
* [x] create & customize `.github/workflows/unittests.yml` d28aa81
* [x] create all environments within `ci/` (consult old `.travis.yml`)
  * [x] pip / git envs 0a5f243
  * [x] standard / plus env 0a5f243
  * [x] try Python 3.8 0a5f243
  * [ ] ~~try Python 3.9~~ ac5c014 *dependencies not there yet*
* [x] update `README.md` 98fbc0c
* [x] update `setup.py` cc5aade, ac5c014
* [x] initialize `codecov`
* [x] create associated `codecov` files 00de405
  * [x] `.coveragerc`
  * [x] `codecov.yml`

**Comment by jGaboardi** [comment]
Create separate issue for review of doctests [due to failures](https://github.com/pysal/spreg/runs/1631382779?check_suite_focus=true#step:9:311).

**Comment by jGaboardi** [comment]
Small precision difference in Windows vs. Linux systems result [in failed tests](https://github.com/pysal/spreg/pull/55/checks?check_run_id=1631422401#step:9:457). Fix these and round back on Windows testing support later.

**Comment by jGaboardi** [comment]
[Cryptic errors on Windows](https://github.com/pysal/spreg/pull/55/checks?check_run_id=1631422401#step:9:380):

```
_____________________ Test_Panel_Diagnostics.test_Hausman _____________________
381
382 self = <spreg.tests.test_diagnostics_panel.Test_Panel_Diagnostics testMethod=test_Hausman>
383
384    def test_Hausman(self):
385          fe_lag = Panel_FE_Lag(self.y, self.x, self.w)
386          fe_error = Panel_FE_Error(self.y, self.x, self.w)
387          re_lag = Panel_RE_Lag(self.y, self.x, self.w)
388 >       re_error = Panel_RE_Error(self.y, self.x, self.w)
389
390 spreg\tests\test_diagnostics_panel.py:57: 
391 _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
392 spreg\panel_re.py:692: in __init__
393    BasePanel_RE_Error.__init__(self, bigy, bigx, w, epsilon=epsilon)
394
spreg\panel_re.py:542: in __init__
395    vm1 = np.linalg.inv(v)
396 <__array_function__ internals>:6: in inv
397    ???
398
C:\Miniconda3\envs\test\lib\site-packages\numpy\linalg\linalg.py:546: in inv
399    ainv = _umath_linalg.inv(a, signature=signature, extobj=extobj)
400 _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
401
402 err = 'invalid value', flag = 8
403
404    def _raise_linalgerror_singular(err, flag):
405 >       raise LinAlgError("Singular matrix")
406 E       numpy.linalg.LinAlgError: Singular matrix
```

**Comment by jGaboardi** [comment]
Dependencies for testing with Python 3.9 are not yet available (see ac5c014).
